File: modules/components/spring-boot-starter-minio/src/main/java/org/toolkit/spring/boot/starter/minio/configurations/MinioConfigurationProperties.java
package org.toolkit.spring.boot.starter.minio.configurations;

import java.util.concurrent.TimeUnit;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;
import org.springframework.boot.context.properties.ConfigurationProperties;

@ConfigurationProperties(prefix = "toolkit.minio", ignoreInvalidFields = true)
@ToString
@Getter
@Setter
public class MinioConfigurationProperties {
	private String previewPattern = "/minio/preview/**";

	private long credentialExpire = 1;

	private TimeUnit credentialUnit = TimeUnit.MINUTES;

	private String endpoint;

	private String accessKey;

	private String secretKey;

	private String defaultBucket;

	private String instancePublicAccessUrl;
}

File: spring-boost-minio/src/main/kotlin/org/spring/boost/minio/autoconfigure/MinioConfigurationProperties.kt
package org.spring.boost.minio.autoconfigure

import org.springframework.boot.context.properties.ConfigurationProperties

data class MinioClientConfig(
    private var endpoint: String? = null,

    private val accessKey: String? = null,

    private val secretKey: String? = null,

    private val defaultBucket: String? = null,
)

@ConfigurationProperties("spring.boost")
data class MinioConfigurationProperties(
    private var endpoint: String? = null,

    private val accessKey: String? = null,

    private val secretKey: String? = null,

    private val defaultBucket: String? = null,

    private val clients:Map<String,MinioClientConfig>? = null
) {
}